Remove zebra striped rows from datatables

krauselukas opened the pull request:
People with limited color vision have issues to read
colored text/items on the stronger gray of the zebra striped
tables.

Fixes #8116

**Before**
![Screenshot-2019-8-21 Open Build Service(1)](https://user-images.githubusercontent.com/22001671/63432825-37772c80-c422-11e9-8297-5ae122149747.png)

**After**
![Screenshot-2019-8-21 Open Build Service](https://user-images.githubusercontent.com/22001671/63432837-4067fe00-c422-11e9-812a-6a568e2e1f51.png)

I would suggest to make the zebra effect a bit lighter. Incase people still having issues to read the content, we still can remove it completly.

Here some interesting article regarding this topic:
https://alistapart.com/article/zebrastripingdoesithelp/

I added a hover effect to the datatables, this makes it easier to keep track of the currently selected row.

hennevogel commented on src/api/app/assets/stylesheets/webui2/bootstrap_variables/tables.scss:
```diff
@@ -1,2 +1,6 @@
-$table-accent-bg: $gray-300;
+$table-accent-bg: $gray-100;
+
+table.dataTable tbody tr:hover {
```
Why not [`.table-hover`](https://getbootstrap.com/docs/4.3/content/tables/#hoverable-rows)?

dmarcoux commented:
hover won't work on mobile.

hennevogel commented on src/api/app/assets/stylesheets/webui2/bootstrap_variables/tables.scss:
```diff
@@ -1,2 +1,6 @@
-$table-accent-bg: $gray-300;
+$table-accent-bg: $gray-100;
+
+table.dataTable tbody tr:hover {
```
Why not [`.table-hover`](https://getbootstrap.com/docs/4.3/content/tables/#hoverable-rows)?

krauselukas replied:
> Hover won't work on mobile.

@dmarcoux I will check if there is a solution for mobile

krauselukas replied:
@hennevogel not sure what you mean exactly, there is not class ```.table-hover```?

dmarcoux replied:
Instead of having this custom rule, `.table-hover` can be used.

krauselukas replied:
Ah okay, get it, okay but I will wait until we agree on having it, since i need to add this class to many tables :D

krauselukas commented:
> Hover won't work on mobile.

@dmarcoux I will check if there is a solution for mobile

krauselukas commented:
I removed the zebra stripes completly, since it doesn't really provides a benifit regarding the readability, and it makes it even harder to distinguish between the elements for people with limited color vision.

I moved the hover part out of this PR and will open another one for this.

**Tables without the zebra stripes**
![Screenshot-2019-8-22 Open Build Service(1)](https://user-images.githubusercontent.com/22001671/63513061-889c2480-c4e5-11e9-8f2e-cbff4e5f6a56.png)
![Screenshot-2019-8-22 This is a test project](https://user-images.githubusercontent.com/22001671/63513063-8c2fab80-c4e5-11e9-8104-f4057c818fd0.png)

coolo commented:
>  it doesn't really provides a benifit regarding the readability

Do you have any proof of that? Because I for one find it much easier to follow a coloured line

krauselukas commented:
@coolo check out those articles: <https://alistapart.com/article/zebrastripingdoesithelp/> and the follow up <https://alistapart.com/article/zebrastripingmoredataforthecase/>

It's kind of a difficult topic, it seems like it slightly influences the readability, but its a lot about personal preferences. The improvement of the readability is small (or in some cases not even measureable) and on the other hand it makes the experience for another group of people worse. Since the improvement through the zebra stripes is rather small for most of the people, I would go the other way and make it better for them who are suffering from it.
I will also open a PR which introduces highlighting of the row on hover, this should support keeping track of the current position inside the table :)

coolo commented:
Aehm, the articles linked spent a lot of effort to conclude with:

```
The results of the three studies conducted to date suggest that the safest option 
is to shade the alternating, individual rows of your table with a single color. 
```

So while I understand there are people who have a problem with the stripes, I can read very well out of the linked articles that I'm not alone with having a problem in removing it.

coolo commented:
I actually found the light option the best - also for reading, not as a compromise :)

hennevogel commented:
@coolo that is why we are going to bring it back on hover...

adrianschroeter commented:
@krauselukas good idea ... could you also bring back the former meaning of the inverted background? (I guess it was only dropped because of the new strips).

The inverted background was used on all states which are still in flux (supposed to change). These are scheduled/building/finished/blocked/dispatched/signing out of my head (but there used to be a variable to define these).

This is really helpful in some more complex projects on the monitor page, it takes atm additional time to group these in mind (and having the active background only on "blocked" state is quite meaningless)

krauselukas commented:
> @krauselukas good idea ... could you also bring back the former meaning of the inverted background? (I guess it was only dropped because of the missing contrast with the new stripes).
> 
> The inverted background was used on all states which are still in flux (supposed to change). These are scheduled/building/finished/blocked/dispatched/signing out of my head (but there used to be a variable to define these).
> 
> This is really helpful in some more complex projects on the monitor page, it takes atm additional time to group these in mind (and having the active background only on "blocked" state is quite meaningless)

@adrianschroeter yes I can bring it back, I will check how much work it is, and depending on that adding it to this PR or to another one

krauselukas commented:
@hennevogel @adrianschroeter I will tackle the background for states that are still in flux in another PR
